### PR TITLE
Added single character selection screen and pause menu

### PR DESCRIPTION
**Pull request by trudeaua**
### New stuff
- Added a single menu for selecting characters and classes
- Added a prefab for navigating and controlling GUIs called `PlayerUIControl`
- Added pools of textures and prefabs to `PlayerManager.cs`
- Added a list to keep track of joined player in `PlayerManager.cs` along with a class called `Player` which stores a joined player's `PlayerInput` (input device, control scheme, etc.), player number, class choice, character choice, and their UI control object
- Players join by pressing X if on a PS4 gamepad, the "south" button if on another gamepad, or any keyboard button if on keyboard/mouse
- If playing on a PS4 gamepad, the [lightbar](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRWRV2AH8d0ZGDBkTeEWA9ceOOFrZtdSchWMAO-4Sq3hARKDuajNw&s) changes colour to match the player's assigned colour (purple, green, yellow, red) **For this reason and for the purpose of looking cool I think it's best to use PS4 controllers during the final demo, and that we should put controls/instructions in terms of PS4 controls**
- Players can now navigate back to the previous menu from any menu (PS4 circle button or the "east" button on other gamepads when on character selection screen, "Back" button on every other screen)
- Player models are shown and updated as you select different characters and classes
- Changed the font of the menu titles to match the buttons
- Turned the volume down on the music
- Disabled player input when players are spawning
- Pressing `Exit Game` in the main menu now exits the game

### TL;DR
Fixed/addressed #135 , #202, #162, #138, #213, #111, #81, #54, and #53

### CODE REVIEW

File: Cosmic Quest - Order and Chaos/Assets/Scripts/Menu/StepperControl.cs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+{
+    public enum StepperProperty { 
+        Character,
+        Class
+    }
+    public int playerNumber;
+    public StepperProperty property;
+    public GameObject playerContainer;
+    private string[] classNames = { "Mage", "Melee", "Healer", "Ranger" };
+    private string[] characterNames = { "Alien A", "Alien B", "Alien C", "Robot" };
```
Comment (camvr) [inline Cosmic Quest - Order and Chaos/Assets/Scripts/Menu/StepperControl.cs]: In the future (if we have time) we should use a slightly more "manageable" way for handling character textures, so that if we wanted to add more into the game, we would only have to add them into a prefab or something

File: Cosmic Quest - Order and Chaos/Assets/Scripts/Game/GameSceneManager.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
+﻿using System.Collections;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class GameSceneManager : MonoBehaviour
```
[the remark this answers is not in the document]
Comment (trudeaua) [reply Cosmic Quest - Order and Chaos/Assets/Scripts/Game/GameSceneManager.cs]: Sounds good!

File: Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerManager.cs [hunk 3]
```diff
@@ -43,9 +94,24 @@ private void Awake()
     // TODO change this to a pool of textures, or assigned to a player at class selection
     public Texture testPlayerTexture;
 
+    // Pool of textures
+    [SerializeField] private Texture[] texturePool;
+    private static Texture[] TexturePool;
+
+    // Pool of player prefabs
+    [SerializeField] private GameObject[] playerPrefabPool;
+    private static GameObject[] PlayerPrefabPool;
```
[the remark this answers is not in the document]
Comment (trudeaua) [reply Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerManager.cs]: the non-static pool of textures is set in the inspector, and the static pool of textures is used in static methods used for assigning/looking up textures . I couldn't think of a way to access a non-static class from a static method and static properties can't be set in the inspector. Same deal with the prefabs

File: Cosmic Quest - Order and Chaos/Assets/Scripts/Menu/StepperControl.cs (hunk 1 above)
Comment (trudeaua) [reply Cosmic Quest - Order and Chaos/Assets/Scripts/Menu/StepperControl.cs]: I could create a class called "CharacterPool" or something that would have a model prefab, a texture, and a name and create an array of those objects in PlayerManager?

File: Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerManager.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (trudeaua) [reply Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerManager.cs]: @camvr Nevermind! Used the `_instance` property of the player manager to refer to the texture and prefab pools. Just pushed a commit with the change

**Comment by tommy-tran** [comment]
Looks awesome :)

Tutorial isn't working for me, do you get the same errors?
image

Getting a similar error on first level too
image

File: Cosmic Quest - Order and Chaos/Assets/Scripts/Menu/StepperControl.cs (hunk 1 above)
Comment (trudeaua) [reply Cosmic Quest - Order and Chaos/Assets/Scripts/Menu/StepperControl.cs]: @camvr I made 2 arrays (set in the inspector) in [my latest commit](https://github.com/olejarza/VACATe/pull/244/commits/0633cd524da51abc20333ecdc445a8f3ba7074cd) that manages class and character choices

File: Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerManager.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (camvr) [reply Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerManager.cs]: Ahh yeah that makes sense why you did that then... Probably should have made the singleton pattern slightly more obvious for that class. Thanks!

File: Cosmic Quest - Order and Chaos/Assets/Scripts/Menu/StepperControl.cs (hunk 1 above)
Comment (camvr) [reply Cosmic Quest - Order and Chaos/Assets/Scripts/Menu/StepperControl.cs]: Awesome! I'll check it out 🙂

File: Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerManager.cs [hunk 4]
```diff
@@ -70,10 +71,24 @@ public Player(PlayerInput _playerInput, CharacterColour _characterColour, Charac
...
+{
+    public string name;
+    public Texture skin;
+}
+
 public class PlayerManager : MonoBehaviour
 {
     #region Singleton
-    private static PlayerManager _instance;
+    public static PlayerManager _instance;
```
Comment (camvr) [inline Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerManager.cs]: Slightly nit-picking here... But could you change the variable name to `Instance`?

For whatever reason I decided to break my own singleton implementation convention with this class, hence why some of the functions are static. I'll probably look into fixing that in my next PR.

Comment (trudeaua) [reply Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerManager.cs]: Sure will do! Thanks @camvr
